### PR TITLE
Attempt at `torch.bucketize`

### LINKED ISSUE #19593: bucketize

### PR DESCRIPTION
**Pull request by benclarkegithub**
Close #19593

My tests are failing, it seems that `boundaries` is being generated with 0 dimensions even though I have set `min_num_dims=1` and `max_num_dims=1`. What can I do to fix it?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 1]
```diff
@@ -1557,3 +1557,46 @@ def test_torch_cov(
...
+    right,
+    test_flags,
+    backend_fw,
+    on_device,
+    frontend,
+    fn_tree,
+):
+    dtypes_input, input = dtypes_and_input
+    dtypes_boundaries, boundaries = dtypes_and_boundaries
+    boundaries[0] = np.sort(boundaries[0])
```
Comment (bipinKrishnan) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1590]: This line might be causing the issue, modify this to `np.sort(boundaries[0], axis=0)`

Comment (benclarkegithub) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1590]: Thanks for the quick reply. I tried that and unfortunately the tests still fail. Printing `boundaries` I can see that sometimes arrays of dimension <1 and >1 are produced, e.g.
```
boundaries: [array(6, dtype=uint8)]
boundaries: [array([[[[[-6.9198933e+16],
          [ 1.9000000e+00]]],


        [[[ 5.9604645e-08],
          [ 1.0000000e+00]]],


        [[[ 1.1920929e-07],
          [-3.0934760e+38]]],


        [[[-2.0000100e+00],
          [-7.3511812e+15]]],


        [[[-3.0934760e+38],
          [-1.0000000e+00]]],


        [[[-6.5933984e+16],
          [-5.9604645e-08]]],


        [[[ 2.0000100e+00],
          [-5.9604645e-08]]],


        [[[ 1.3810570e+37],
          [ 1.4916630e+16]]],


        [[[ 6.1035156e-05],
          [-2.0000100e+00]]],


        [[[ 3.3333334e-01],
          [ 2.1191212e+38]]]]], dtype=float32)]
```
I believe this is to do with `helpers.dtype_and_values`. Am I misunderstanding the function? I assumed that by setting `min_num_dims=1` and `max_num_dims=1` I would only ever get arrays of dimension 1.

Comment (bipinKrishnan) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1590]: Currently, the issue is raised due to not setting axis=0, can you do the change for this so that we could look into the exception which you mentioned.

**Comment by bipinKrishnan** [comment]
Hey @benclarkegithub, can you also fix the conflicts, since you are directly working on the master of your fork, I may not have push access to fix the conflicts. Thanks :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py (hunk 1 above)
Comment (benclarkegithub) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1590]: I added `axis=0`

File: .idea/runConfigurations/_template__of_py_test.xml
Comment (bipinKrishnan) [inline .idea/runConfigurations/_template__of_py_test.xml:1]: These might not be related to your PR, please undo these changes

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
